src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -476,38 +476,33 @@ private bool IsParamsArrayExpanded(SemanticModel semanticModel, SyntaxNode node,
                     ObjectCreationExpressionSyntax objectCreation => objectCreation.ArgumentList,
                     ConstructorInitializerSyntax constructorInitializer => constructorInitializer.ArgumentList,
                     ElementAccessExpressionSyntax elementAccess => elementAccess.ArgumentList,
-                    _ => throw new ArgumentException("Unexpected SyntaxNode", nameof(node))
+                    _ => throw ExceptionUtilities.UnexpectedValue(node.Kind())
                 };
 
                 argumentCount = argumentList.Arguments.Count;
                 lastArgumentHasNameColon = argumentList.Arguments.LastOrDefault()?.NameColon != null;
                 lastArgumentExpression = argumentList.Arguments.LastOrDefault()?.Expression;
             }
 
-
             if (symbolInfo.Symbol is IMethodSymbol methodSymbol && methodSymbol.Parameters.LastOrDefault()?.IsParams == true)
             {
-                if (argumentCount > ((IMethodSymbol)symbolInfo.Symbol).Parameters.Length)
+                if (argumentCount > methodSymbol.Parameters.Length)
                 {
                     return true;
                 }
-                else if (argumentCount == ((IMethodSymbol)symbolInfo.Symbol).Parameters.Length)
+                else if (argumentCount == methodSymbol.Parameters.Length)
                 {
                     if (lastArgumentHasNameColon)
                     {
-                        return true;
+                        return false;
```

Comment:
=-o   intentional change?

Reply:
Yes. You pointed out that it didn't make sense, and you were right. I fixed it and added a test. 😬😬😬